### PR TITLE
Added support to suggest package name for `elm-package diff` command.

### PR DESCRIPTION
**Pull request by akoppela**
Elm package diff command also requires package names for completion.

### CODE REVIEW

File: _elm_package.sh [hunk 1]
```diff
@@ -169,6 +169,9 @@ _elm_package()
         if [[ $cur == -* ]]; then
             _suggest_package_diff_flags $cur
             return 0
+        else
+            _suggest_package_name $cur
```
Comment (eeue56) [inline _elm_package.sh]: I think this might actually be wrong -- what gets suggested when you have `elm-package diff elm-lang/core 1.0.0 [TAB]`? What do you get suggested at `[TAB]`?

Comment (akoppela) [reply _elm_package.sh]: Ah yes, package name get suggested after first version number

Comment (akoppela) [reply _elm_package.sh]: But before there were no suggestions at all, neither for package name nor for version numbers. E.g. `elm-package install elm-lang/core [TAB]` will also suggest name packages.

Comment (eeue56) [reply _elm_package.sh]: maybe you can figure out howto fix that as part of this pr?

Comment (akoppela) [reply _elm_package.sh]: What do you think about checking the word index but excluding indexes for flags? It should always be the same number.
E.g.
In `elm-package install package/name` `package/name` will have index `2`. In `elm-package install --yes package/name` `package/name` will have index `3` minus `1` for `--yes` tag which will be again `2`.

So if we check that current index minus number of flags equals to `2` than we do `_suggest_package_name $cur`. Otherwise do nothing.

Comment (akoppela) [reply _elm_package.sh]: Or do you mean auto completion for version numbers?

Comment (akoppela) [reply _elm_package.sh]: I've added the function, now it works as expected. What you you think?

File: _elm_package.sh [hunk 2]
```diff
@@ -95,8 +95,31 @@ _suggest_package_publish_flags()
     COMPREPLY=( $(compgen -W "${flag_options}" -- $cur) )
 }
 
+_is_suggesting_package_name()
+{
+    local flags_before_package_name collect_flags
+
+    flags_before_package_name=0
+    collect_flags=1
```
Comment (eeue56) [inline _elm_package.sh:103]: Why is this needed?

Comment (akoppela) [reply _elm_package.sh:103]: There are two situations
• When flags are before package name `elm-package install --flag1 --flag2 ... package/name`
• When flags are after package name `elm-package install package/name --flag1 --flag2 ...`

We want to count flags which are before package name to exclude them. So that package name will have always the same index. Which equals to `2`.

**Comment by akoppela** [comment]
Nice :) I’ve noticed that `elm-help` also uses package names. Do you want me to implement the same for `elm-help`?

**Comment by akoppela** [comment]
Ah no. For `elm-help` there is a different logic. Auto completion for package name will only happen if previous word is `-p` or `--package` flag. So there is ok.
